src/main/java/com/fincons/service/importFile/impl/PersistenceEmployeeServiceImpl.java

```diff
@@ -1,33 +1,87 @@
 package com.fincons.service.importFile.impl;
 
+import com.fincons.controller.UserController;
 import com.fincons.dto.EmployeeDTO;
 import com.fincons.dto.ErrorDetailDTO;
+import com.fincons.entity.Employee;
+import com.fincons.entity.User;
 import com.fincons.enums.ErrorCode;
+import com.fincons.exception.EmailException;
+import com.fincons.repository.UserRepository;
+import com.fincons.service.authService.UserServiceImpl;
 import com.fincons.service.employeeService.EmployeeService;
 import com.fincons.service.importFile.PersistenceEmployeeService;
 import com.fincons.utility.EmployeeMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.sql.SQLException;
+import java.sql.SQLIntegrityConstraintViolationException;
 import java.util.ArrayList;
 import java.util.List;
-
  @Service
 public class PersistenceEmployeeServiceImpl implements PersistenceEmployeeService {
 
     @Autowired
     EmployeeService employeeService;
+
+    
+    @Autowired
+    UserServiceImpl userService;
+
+
     @Override
-    public List<ErrorDetailDTO> addIfNotPresent(List<EmployeeDTO> employeeList) {
-        List<ErrorDetailDTO> duplicatedResultList= new ArrayList<>();
-        for (EmployeeDTO employee: employeeList) {
+    @Transactional
+    public void addIfNotPresent(EmployeeDTO employee, List<ErrorDetailDTO> duplicatedResultList) throws RuntimeException{
+        int contEmployee=0, contUser=0;
             long nRow= employee.getRowNum();
             boolean employeeExists= employeeService.employeeExists(EmployeeMapper.convertToEntity(employee));
-            if(employeeExists){
-                duplicatedResultList.add(new ErrorDetailDTO(nRow, "EmailId: " + employee.getEmail(), ErrorCode.RESOURCE_ALREADY_EXISTS));
+                if(!employeeExists){
+                    Employee employeeEntity= EmployeeMapper.convertToEntity(employee);
```

Comment:
La conversione in entity viene fatta 2 volte (riga 39), per ottimizzare potrebbe essere fatta 1 sola.